Prepare CG display-list ImageBuffer to contain more structured data

hortont424 opened the pull request:
#### 11fb11c7389fe532c82b8e9b89a4c32bc07b3485
<pre>
Prepare CG display-list ImageBuffer to contain more structured data
<a href="https://bugs.webkit.org/show_bug.cgi?id=243195">https://bugs.webkit.org/show_bug.cgi?id=243195</a>

Reviewed by Chris Dumez.

* Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.cpp: Added.
(WebKit::CGDisplayList::CGDisplayList):
(WebKit::CGDisplayList::encode const):
(WebKit::CGDisplayList::decode):
* Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.h: Added.
(WebKit::CGDisplayList::buffer const):
* Source/WebKit/Shared/RemoteLayerTree/CGDisplayListImageBufferBackend.cpp:
(WebKit::CGDisplayListImageBufferBackend::createBackendHandle const):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStore::applyBackingStoreToLayer):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/ImageBufferBackendHandle.h:
Factor out CGDisplayList instead of using SharedBuffer directly,
in preparation for it carrying more than just a single blob of data.

Canonical link: <a href="https://commits.webkit.org/252818@main">https://commits.webkit.org/252818@main</a>
</pre>

cdumez commented on Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.cpp:
```diff
@@ -0,0 +1,53 @@
+
+namespace WebKit {
+
+CGDisplayList::CGDisplayList()
```
= default; ?

hortont424 replied:
Moved this back to the header and yes.

cdumez commented on Source/WebKit/Shared/RemoteLayerTree/CGDisplayListImageBufferBackend.cpp:
```diff
@@ -108,7 +109,9 @@ ImageBufferBackendHandle CGDisplayListImageBufferBackend::createBackendHandle(Sh
 
-    return ImageBufferBackendHandle { IPC::SharedBufferReference { WebCore::SharedBuffer::create(data.get()) } };
+    CGDisplayList displayList;
+    displayList.m_displayList = WebCore::SharedBuffer::create(data.get());
```
Why set a private data member like this instead of passing the display list to the constructor?

hortont424 replied:
Fixed! This was a silly thing I inherited from ShareableBitmap::Handle.

cdumez commented on Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.h:
```diff
@@ -0,0 +1,57 @@
+    WTF_MAKE_NONCOPYABLE(CGDisplayList);
+public:
+    CGDisplayList() = default;
+    CGDisplayList(WebCore::SharedBuffer& displayList)
```
Did you intentionally not mark it as explicit?

cdumez reviewed (approved):
LGTM

hortont424 committed:
Prepare CG display-list ImageBuffer to contain more structured data
https://bugs.webkit.org/show_bug.cgi?id=243195

Reviewed by Chris Dumez.

* Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.cpp: Added.
(WebKit::CGDisplayList::CGDisplayList):
(WebKit::CGDisplayList::encode const):
(WebKit::CGDisplayList::decode):
* Source/WebKit/Shared/RemoteLayerTree/CGDisplayList.h: Added.
(WebKit::CGDisplayList::buffer const):
* Source/WebKit/Shared/RemoteLayerTree/CGDisplayListImageBufferBackend.cpp:
(WebKit::CGDisplayListImageBufferBackend::createBackendHandle const):
* Source/WebKit/Shared/RemoteLayerTree/RemoteLayerBackingStore.mm:
(WebKit::RemoteLayerBackingStore::applyBackingStoreToLayer):
* Source/WebKit/SourcesCocoa.txt:
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:
* Source/WebKit/WebProcess/GPU/graphics/ImageBufferBackendHandle.h:
Factor out CGDisplayList instead of using SharedBuffer directly,
in preparation for it carrying more than just a single blob of data.

Canonical link: https://commits.webkit.org/252818@main
